From 84be6e8cd9136be35f11709c09694cc03f129c77 Mon Sep 17 00:00:00 2001
From: Fabio Bertinatto <fbertina@redhat.com>
Date: Thu, 27 Apr 2023 09:46:33 -0300
Subject: [PATCH] UPSTREAM: <carry>: Export internal code from
 k8s.io/apimachinery/pkg/util/managedfields

Some of the code we use in openshift-tests was recently made internal
in https://github.com/kubernetes/kubernetes/pull/115065. This patch
exposes the code we need there.
---
 .../apimachinery/pkg/util/managedfields/patch.go  | 15 +++++++++++++++
 1 file changed, 15 insertions(+)
 create mode 100644 staging/src/k8s.io/apimachinery/pkg/util/managedfields/patch.go

diff --git a/staging/src/k8s.io/apimachinery/pkg/util/managedfields/patch.go b/staging/src/k8s.io/apimachinery/pkg/util/managedfields/patch.go
new file mode 100644
index 00000000000..ae77235ae18
--- /dev/null
+++ b/staging/src/k8s.io/apimachinery/pkg/util/managedfields/patch.go
@@ -0,0 +1,15 @@
+package managedfields
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/managedfields/internal"
+)
+
+// ManagedInterface groups a fieldpath.ManagedFields together with the timestamps associated with each operation.
+type ManagedInterface = internal.ManagedInterface
+
+// DecodeManagedFields converts ManagedFields from the wire format (api format)
+// to the format used by sigs.k8s.io/structured-merge-diff
+func DecodeManagedFields(encodedManagedFields []metav1.ManagedFieldsEntry) (ManagedInterface, error) {
+	return internal.DecodeManagedFields(encodedManagedFields)
+}
-- 
2.46.0

